### ISSUE TITLE
Training a neural net in C++

### ISSUE THREAD

**Issue by sonalgupta** [tensorflow/tensorflow#4130]
1. There are examples in the repo for loading a pre-trained model in C++. Can someone add an example to train a simple 1 hidden layer fully connected neural net (or even logistic regression using SGD) in C++? 
2. I realize there is no publicly available support for auto differentiation in C++. Are there any plans to do so?

**Comment by keveman** [comment]
@sonalgupta [This](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/cc/tutorials/example_trainer.cc) is an example that constructs a graph is C++ and runs many iterations of it to solve a numerical optimization problem. That example is missing automatic differentiation which is the missing component for training a neural net model, but @andydavis1 is making rapid progress towards that. Stay tuned.

**Comment by drpngx** [comment]
@jhseu and @asimshankar for additional comments.

**Comment by asimshankar** [comment]
Broadly speaking, there are two things holding training in languages other than python up. One is support for gradients (#6268) and the other is the fact that the "control logic" for training (optimizers, estimators etc.) are in python.

#6268 talks about covering gradients. Once that is is resolved (and that might be a while), we'll have enough functionality in these languages to build up the control logic.

With that said, I'm tempted to close this out as a duplicate of #6268.
